### DISCUSSION TITLE
v2.0.0-beta.28

### DISCUSSION THREAD

**Discussion by 0xJacky** [0xJacky/nginx-ui#450] [Announcements]
# Notice
We're excited to announce that we've been hard at work developing the next generation of the Nginx UI (v2). To keep up-to-date with our progress, please visit https://github.com/users/0xJacky/projects/1.

Please feel free to send any feedbacks, suggestions or bug reports in Github issues.

# Change vs v2.0.0-beta.27
1. Hide some parts of Jwt Secret and Node Secret in settings page.
2. The selector of change certificate now supports selecting multiple certificates #437 .
3. Certificate(s) Info now supports display info of multiple certificates.
4. The info of changed certificates in server tab will be shown. #386
5. Fix a bug of the loading indicator constantly spinning in the table of configurations management page, if the directory has no file.
6. Better breadcrumbs #258.
7. Add OCSP Must Staple options in AutoCert #292.
8. Add LEGO_DISABLE_CNAME_SUPPORT options in AutoCert #407.
9. Fix node selector styles broken in medium/small width screen.

# Breaking Changes:
1. About PR #445, if you are using nginx-ui docker container and encounter this issue, you have to remove the trailing `.conf` manually. Diff: https://github.com/0xJacky/nginx-ui/pull/445/files
5. The `nginx_log` section in the configuration file has been renamed to `nginx`.
`NginxConfigDir` and `NginxPIDPath` in the `server` section have been renamed to `ConfigDir` and `PIDPath` respectively, and both have been moved into the `nginx` section.
8. The `ReloadCmd` and `RestartCmd` have been added in the `nginx` section to configure custom commands for reloading/restarting Nginx.
11. In versions above v2.0.0-beta.23 of Nginx UI Docker containers, `/var/logs/nginx/access.log` and `/var/logs/nginx/error.log` are no longer soft links to `/dev/stdout` and `/dev/stderr`, but regular files. If you need this functionality, please add `access_log /dev/stdout` and `error_log /dev/stderr` in your `nginx.conf`. For users upgrading from older versions who do not need this feature, check if your `nginx.conf` contains `access_log /var/logs/nginx/access.local.log` and `error_log /var/logs/nginx/error.local.log`. If so, please manually remove them to avoid duplicate log entries. This change allows you to map `/var/logs/nginx` to a directory on the host machine to achieve persistent storage of log files.

# What's new:
1. We're proud to have registered the [nginxui.com](https://nginxui.com) domain 🎉.
12. We have launched a new documentation site, please check [nginxui.com](https://nginxui.com). 
13. Node management functionality has been introduced (#70).
14. Duplicate site function now can duplicate config to remote server.
15. French translations have been added, thanks to the contributions of @skrilax91.
16. Added border-radius to Ace Editor and terminal, we hope you will like this change :).
17.  You can now deploy site configuration to a remote server.
18. Spanish translations have been added, thanks to the contributions of @kcholoren.
19. Russian translations have been added, thanks to the contributions of @iAHTOH.
20. The dashboard now includes a system monitor for nodes.
21. Use jsoniter for improving gin performance.
22. Added custom reload/restart command for nginx. #140
23. Refacted translation module #135 
24. Added http host config #157 by @quanbisen 
25. Add setting for test nginx config command #191 
26. Added casdoor sso support, thanks to the contributions of @Jraaay 
27. Refactor the project directory structure.
28. Use tailwind css.
29. Use eslint to check the code style of frontend.
30. Use vue-tsc to check the types for frontend.
31.  Introducing a new management page for certificates, user can now view the latest renew logs of the certain certificate.
32. Add manually renew button in certificate modify page for managed certificate (auto cert).
33. Add notifications for renew certificates. #192 
34. Automatically detect the path of accress logs and error logs.
35. Introducing the feature of issuing wildcard certificates on the Manage Certificate page. #190 
36. Add some Nginx Config Templates #233, thanks to the contributions of @sanvu88 . 
37. Support different key types for obtaining certificate, including RSA2048, RAS3072, RSA4096, EC256 and EC384 #264.
38. Introducing logroate cron task for docker users #255.
39. A system default acme user will be registered when nginx ui start. https://github.com/0xJacky/nginx-ui/issues/237
40. Add acme user management page. https://github.com/0xJacky/nginx-ui/issues/319
41. Persist the acme user to the database.
42. Add a setting of renewal interval of certificates (7-21d). https://github.com/0xJacky/nginx-ui/issues/343
43. Add a option to select a acme user when obtaining a certificate. https://github.com/0xJacky/nginx-ui/issues/319 https://github.com/0xJacky/nginx-ui/issues/16
44. SSL management support different types of certificates of a same doamin name. https://github.com/0xJacky/nginx-ui/issues/309, https://github.com/0xJacky/nginx-ui/issues/288
45. Certificates obtained within 21 days are renewed by using the renewal api. For compatibility, the challenge resolver is still setup when renewing. https://github.com/0xJacky/nginx-ui/issues/319
46. Add trashs for soft deleted items in CURD pages.
47. Use s6-overlay to manage the services in docker #308 .
48. Users can use settings or environment variable to skip installation #357 .
49. Users can use environment variable to predefine a admin user  (only available in skip installation mode) #214 .
50. Users can customize local environment name #313 .
51. Users can use settings file to predefine nodes #169 .
52. Hot reload cluster node settings from settings file #169.
53. Introducing synchronize certificates to remote server #293, #363, #411.
54. Enabled search box for Ace code edior #412.
55. Add a white list of IPs, only those in list can access the Nginx UI. 
56. Add login failed ban IP list. 
57. Add 2FA authorization for login #344, #419.
58. Add 2FA authorization for Web Terminal #394 .

# Fixes:
1. We've upgraded gonginx to address the configuration parsing issue #73.
10. Fixed an issue where it was not possible to update settings value to an empty string.
11. Resolved a location saving issue on the domain addition page #116.
12. Site enabling/disabling is now aborted only when the log level is higher than warning.
13. Fixed environment indicator display overlay and data obtaining issues #152 by @quanbisen 
14. Fixed env indicator background in dark mode.
15. Terminal process not killed #181 
16. Added `gpt-4-1106-preview` model in ChatGPT settings.
17. Fixed change language cause unexpected redirect #189 
18. Fixed english grammar error in button text in sites management page. #200 
19. Fixed logs mnual link error. #201 
20. Revert enable site if you trying to enable a incorrect config of the site. #199 
21. Upgrade ant design vue verison from v3 to v4, thanks @funcfang
22. Fixed FooterToolBar content was covered by Monica, a Chrome Extension. #195 
23. Delete server confirm modal not destroy #206 
24. clientMaxBodySize name is wrong. #216 
25. Fix error maybe occurs when enable TLS. #230 
26. Introducing streams management pages. #166 
27. Introducing upstream editor. #138 
28. Introducing upstream health check (supports TCP and Unix Socket).
29. Fix a bug that may casue nginx-ui taking 100% CPU if the log file is not a regular file.

# Enhances:
1. The recovery function now can print call stack information.
8. Added file info to config editor.
9. Improved zh_TW locale. #148 
10. Enhance error handler for execShell.
11. Enhance nginx control and status acquisition. #222 
12. Remove the vaildated rules for DNS Credentials

# Docs:
1. Traditional Chinese documentation has been added, thanks to @Hintay.
7. We've added documentation for the syntax of the configuration template.
8. We have updated the documentation for the nginx proxy example.
13. Updated deprecated TLS directives #183 

# Others:
1. Core upgrades will be performed as a dry run in demo mode.
6. Updated zh_CN translations.
7. Changed Token to NodeSecret #152 
9. Updated configurations of DNS challenges provider.

<hr /><em>This discussion was created from the release <a href='https://github.com/0xJacky/nginx-ui/releases/tag/v2.0.0-beta.28'>v2.0.0-beta.28</a>.</em>

**Comment by jiriteach** [comment]
Great work on this release. Notice the "View" links do do anything in this latest release.
Seems to be the same on all pages.

<img width="1525" alt="Screenshot 2024-07-25 at 8 24 59 AM" src="https://github.com/user-attachments/assets/2044fa5b-dc85-410d-aa18-b57e62859d9a">

**Reply by 0xJacky** [reply]
This bug is in v2.0.0-beta.27, and has been fixed in v2.0.0-beta.28, I will release it as stable verison today.
